Added addendum to storing data blog to add reading from Sharepoint (#291)
The-Strategy-Unit/data_science: blogs/posts/2024-05-22_storing-data-safely/index.qmd

# and storing it in memory)
parquet_in_memory <- AzureStor::storage_download(
  container,
  src = "newdir/cats.parquet", dest = NULL
)
parq_df <- arrow::read_parquet(parquet_in_memory)

# Delete from Azure container (!!!)
for (blobfile in blob_list$name) {
  AzureStor::delete_storage_file(container, blobfile)
}
```

## Azure Storage in Python

This will use the same environment variables as the R version, just stored in a .env file instead.

We didn't cover this in the presentation, so it's not in the slides, but the code should be self-explanatory.

{{< embed azure_python.ipynb echo=true >}}

## ADDENDUM: Accessing data from SharePoint

SharePoint is a Microsoft product, which is a content/knowledge management tool. Many teams across the NHS use SharePoint for all sorts of file types that need to be preserved or shared within and between teams, but also need to be kept secure.

Accessing SharePoint requires user authentication, which also needs to happen when reading data into R from SharePoint. Before you can access SharePoint content from a site, you first need to 'follow' that site.

To access data on SharePoint,follow these steps:

> 1.  Navigate to the SharePoint page that has the file of interest in it, using your browser.

Reviewer: Whitespace.
```suggestion
1. Navigate to the SharePoint page that has the file of interest in it, using your browser.
```

2.  Click the small star in the top right corner of the window, labelled 'Follow'

![](Follow.jpg)

> 3.  Open your environment file, which you can do by running `r file.edit("~/.Renviron")` within the RStudio console area.

Reviewer: Bit more on env vars/.Renviron. Maybe {usethis} for opening the file?
```suggestion
3. Open (or create) your project's `.Renviron` file, used for [storing environmental variables](https://docs.posit.co/ide/user/ide/guide/environments/r/managing-r.html#renviron). You can do this by running `r usethis::edit_r_environ()` from the R console.
```

> 4.  Save two new environment variables; one is the name of the site (i.e. the name at the top of the browser screen), and the other is the full file path from below the 'Documents' folder to the file itself, including the file type extension. Save and close the .Renviron file then restart your R session (CTRL \> Shift \> F10 on Windows machines)

Reviewer: I think we should suggest names for the variables and then refer to them in the section below with those names.

```suggestion
4. Save two new environment variables to the `.Renviron` file:
    a. `SP_SITE_NAME`: the name of the site (i.e. the name at the top of the browser screen)
    b. `SP_FILE_PATH`: the full file path from below the `Documents/` folder to the file itself, including the file type extension. 
5. Save and close the `.Renviron` file then either restart your R session (CTRL + Shift + F10 on Windows machines) or run `r readRenviron(".Renviron")` to make the new variables available in your session.
```

> 5.  Use the code below to read in an Excel file (xlsx) into R memory.

Reviewer: ```suggestion
5. Use the code below to read in an Excel file (xlsx) into R memory, or adapt it to read other file types.
```

```{r}
#| eval: false
#| echo: true

# Read in the securely saved path and site variables

> sharepoint_site <- Sys.getenv("SITE_NAME")
> 
> template_path <- Sys.getenv("PATH_TO_FILE.xlsx")

Reviewer: To match the suggested var names above.
```suggestion
sharepoint_site <- Sys.getenv("SP_SITE_NAME")
template_path <- Sys.getenv("SP_FILE_PATH")
```
